### PR TITLE
fix(checkbox): Emit event when checkbox's indeterminate value changes

### LINKED ISSUE #2004: [md-checkbox] checked and indeterminate state problem.
#### Bug
I believe this is a bug.

#### What is the expected behavior?
If both checked and indeterminate are true and then checked is set to false it should show indeterminate.

#### What is the current behavior?
If both checked and indeterminate are true and then checked is set to false it should show indeterminate but instead, it shows unchecked.

#### What are the steps to reproduce?
If you use the toggles you will see the state of the checkbox not being correct against the set values.
https://embed.plnkr.co/v7euQHvzVsAFHNcpsEFo

#### Which versions of Angular, Material, OS, browsers are affected?
Latest

### PR DESCRIPTION
**Pull request by tinayuangao**
Fixes #2004 

R: @jelbourn

### CODE REVIEW

File: src/lib/checkbox/checkbox.ts [hunk 1]
```diff
@@ -203,6 +209,7 @@ export class MdCheckbox implements ControlValueAccessor {
       this._transitionCheckState(
           this.checked ? TransitionCheckState.Checked : TransitionCheckState.Unchecked);
     }
+    this.indeterminateChange.emit(this._indeterminate);
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.ts]: Should this event only fire when the value changes?

File: src/lib/checkbox/checkbox.ts [hunk 2]
```diff
@@ -138,6 +138,9 @@ export class MdCheckbox implements ControlValueAccessor {
   /** Event emitted when the checkbox's `checked` value changes. */
   @Output() change: EventEmitter<MdCheckboxChange> = new EventEmitter<MdCheckboxChange>();
 
+  /** Event emitted when the checkbox's `indeterminate` value changes. */
+  @Output() indeterminateChange: EventEmitter<Boolean> = new EventEmitter<Boolean>();
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.ts]: `boolean` lower-case

**Comment by tinayuangao** [comment]
Comments addressed. Please take another look. Thanks!

**Comment by djanesch** [comment]
when will this change released?
@tinayuangao @jelbourn

**Pull request merged**